### PR TITLE
adding focused-window-name to show title of focused window

### PR DESCRIPTION
**Pull request by priestoferis**
This is I think my third pull request ever and the first one on a new function rather then just a tweak so I'm not sure if I did this right ...

Anyway, the script returns the title of the focused window. The short version is not really smart, but at least is short. It can also be used with command line arguments. I'm not sure this feature will be useful for anything, but I coded it with that first and then realized I need to use environment variables if I want a nice config in i3xrocks.

### CODE REVIEW

File: i3xrocks.conf [hunk 1]
```diff
@@ -97,3 +97,10 @@ interval=10
 [time]
 interval=60
 
+# Show title of focused window
+[focused-window-name]
```
Comment (kgilmer) [inline i3xrocks.conf]: I would like to keep this disabled by default until/if we determine it is important enough that all users get this blocket by default.

File: Makefile [hunk 2]
```diff
@@ -15,6 +15,7 @@ install:
 	$(INSTALL) -m0755 -D scripts/cpu_usage $(SCRIPT_TARGET)/cpu_usage
 	$(INSTALL) -m0755 -D scripts/disk $(SCRIPT_TARGET)/disk
 	$(INSTALL) -m0755 -D scripts/iface $(SCRIPT_TARGET)/iface
+	$(INSTALL) -m0755 -D scripts/focused_window_name $(SCRIPT_TARGET)/focused_window_name
```
Comment (kgilmer) [inline Makefile]: NIce!

File: i3xrocks.conf (hunk 1 above)
Comment (priestoferis) [reply i3xrocks.conf]: Wow, github has a lot of features I didn't know about :) Anyway, I pushed a commit that adds a bit more text in comments, but comments out the entire section.

**Comment by priestoferis** [comment]
I disabled the blocklet by default. 

Regarding the label, currently the icon is display by having the `label` section set in `i3xrocks.conf`. Does that have specific drawbacks?

Regarding the pango markup. The idea would be to read with `xrescat` the values set in the config like this:

```
VALUE_FONT=${font:-$(xrescat i3xrocks.value.font "Source Code Pro Medium 13")}
```

and use these values to mark up? As far as I can see there is three of interest:
```
i3xrocks.value.color 
i3xrocks.label.color
i3xrocks.value.font
```

If that's the case I'll push a fix for pango.

**Comment by priestoferis** [comment]
label is still added from the config but I added the formatting now it uses the same fonts as the other blocks

**Comment by kgilmer** [comment]
In testing the script I get:
```
$ /usr/share/i3xrocks/focused-window-name
Traceback (most recent call last):
  File "/usr/share/i3xrocks/focused-window-name", line 30, in <module>
    from i3ipc import Connection, Event
ModuleNotFoundError: No module named 'i3ipc'
```

So, this dependency, i3ipc for python, will have to be expressed as a package dependency to ensure users have the necessary libraries installed before running the script.  I looked around and could not find a Debian package for this, so it will have to be packaged before this PR can be merged.  A similar situation to what's going on with https://github.com/regolith-linux/regolith-i3xrocks-config/pull/11.

In addition, since it's requires a specific dependency, it should probably be in a discrete package.

**Comment by kgilmer** [comment]
Added i3ipc here: https://github.com/regolith-linux/speed-ricer/commit/bc57e8bf9cb5803415e1c7068efd314fdcc80f0f

**Comment by kgilmer** [comment]
For reference: https://github.com/altdesktop/i3ipc-python/issues/139

**Comment by priestoferis** [comment]
Oh yes, sorry about that, I had it on my computer for some time and forgot about it ... I'm not sure how to proceed with this. You might think about including i3ipc by default as it might be used to solve nicely the "have a keybind to open the next workspace" problem as well.

**Comment by kgilmer** [comment]
I think the Debian packaging format will allow me to specify a second binary package in which we can ship this blocklet with the declared dependency of i3ipc-python.  So, I'll work on this issue in the packaging code.

**Comment by kgilmer** [comment]
I will test the blocklet after installing my packaging of i3ipc-python, here: https://launchpad.net/~kgilmer/+archive/ubuntu/speed-ricer/+sourcepub/10773128/+listing-archive-extra

**Comment by priestoferis** [comment]
This is a cosmetic update, LibreOffice puts the name of the app into the title i.e `Example.doc - LibreOffice Writer` which just takes up space since you can already see which window is focused anyway.

**Comment by kgilmer** [comment]
Successfully tested after installing i3ipc-python and python3-xlib:

```
$ ./focused-window-name
<span font_desc="Source Code Pro Medium 13" color = "#D8DEE9" > st </span>
<span font_desc="Source Code Pro Medium 13" color = "#D8DEE9" > st </span>
```

I did not add a required dependency into the i3ipc-python package.  I've building [that change](https://github.com/regolith-linux/speed-ricer/compare/bc57e8bf9cb5...8d34efe3acaa) now, but am ready to merge this in.  Thanks @priestoferis for your awesome new blocket!

**Comment by priestoferis** [comment]
cool thanks

**Pull request merged**